dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/EventChangeLogServiceTest.java

```diff
@@ -463,9 +459,225 @@ void shouldReturnEventPropertiesChangeLogWhenExistingGeometryPointPropertyDelete
             assertPropertyCreate("geometry", "(-11.419700, 8.103900)", geometryChangeLogs.get(1)));
   }
 
-  private void updateDataValue(String event, String dataElementUid, String newValue)
-      throws IOException {
-    TrackerObjects trackerObjects = fromJson("tracker/event_and_enrollment.json");
+  private static Stream<Arguments> provideOrderParamsDesc() {
+    return Stream.of(
+        Arguments.of(
+            "createdAt",
+            SortDirection.DESC,
+            Arrays.asList("25", "20", "15"),
+            Arrays.asList("20", "15", null)),
+        Arguments.of(
+            "username",
+            SortDirection.DESC,
+            Arrays.asList("25", "20", "15"),
+            Arrays.asList("20", "15", null)),
+        Arguments.of(
+            "username",
+            SortDirection.ASC,
+            Arrays.asList("25", "20", "15"),
+            Arrays.asList("20", "15", null)));
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideOrderParamsDesc")
+  void shouldSortChangeLogsByCreatedAtDescWhenOrderingByDateOrUsername(
```

Comment:
Is this test supposed to assert the secondary sort on createdAt? as the current/previousValues which I assume are the expected ones are all the same for every test in here.

Reply:
Yes. The username is always the same, so this case will test the secondary sort.